src/cml.js

```diff
@@ -330,11 +330,19 @@ class CML {
   }
 
   async ci(opts = {}) {
-    const { userEmail = GIT_USER_EMAIL, userName = GIT_USER_NAME } = opts;
+    const {
+      unshallow = false,
+      userEmail = GIT_USER_EMAIL,
+      userName = GIT_USER_NAME
+    } = opts;
 
     const driver = getDriver(this);
-    const command = await driver.updateGitConfig({ userName, userEmail });
-    await exec(command);
+    await exec(await driver.updateGitConfig({ userName, userEmail }));
+    if (unshallow) {
+        if ((await exec('git rev-parse --is-shallow-repository')) === 'true') {
```

Comment:
`--is-shallow-repository` is only available in git >= 2.15 (late 2017), not sure if this will cause issues for CML or not (depends on whether or not this is ever run in older/outdated linux distros)

Reply:
Ubuntu 18.04 uses Git [2.17](https://packages.ubuntu.com/bionic/git). I guess it's fine. 🤔

Reply:
Yup, [2.15 is EOL](https://github.com/iterative/cml/pull/957#discussion_r849124319) so should be fine.

Reply:
Makes sense, just thought it's worth mentioning though since I'm pretty sure in redhat/centos land there's still non-eol'd distros that come with git 1.x as the only officially supported git package 🙃

Reply:
Yes, déjà lu: https://github.com/iterative/cml/pull/957#discussion_r849122106, but probably not worth the hassle (?) 😅